sympy/series/fourier.py

```diff
@@ -0,0 +1,391 @@
+
+    References
+    =========
+
+    .. [1] mathworld.wolfram.com/FourierSeries.html
+    """
+    def __new__(cls, func, limits=None, **kwargs):
+        func = sympify(func)
+
+        limits = _process_limits(func, limits)
+        x, lower, upper = limits
+
+        if x not in func.free_symbols:
+            return func
+
+        kan = kwargs.pop('an', None)
+        kbn = kwargs.pop('bn', None)
+
+        if kan is None and kbn is None:
+            n = Dummy('n')
+            neg_func = func.subs(x, -x)
+            if func == neg_func:
+                an = fourier_cos_seq(func, limits, n)
+                bn = SeqFormula(0, (1, oo))
+            elif func == -neg_func:
+                an = SeqFormula(0, (1, oo))
+                bn = fourier_sin_seq(func, limits, n)
+            else:
+                an = fourier_cos_seq(func, limits, n)
+                bn = fourier_sin_seq(func, limits, n)
+        else:
+            an, bn = kan, kbn
+
+        a0 = kwargs.pop('a0', None)
+        if a0 is None:
+            a0 = an[0] / 2
+
+
+        obj = Expr.__new__(cls, func, limits)
+        obj.a0, obj.an, obj.bn = a0, an, bn
+        return obj
+
+    @property
+    def function(self):
+        return self.args[0]
+
+    @property
+    def x(self):
+        return self.args[1][0]
+
+    @property
+    def period(self):
+        return (self.args[1][1], self.args[1][2])
+
+    @property
+    def interval(self):
+        return Interval(0, oo)
+
+    @property
+    def start(self):
+        return self.interval.inf
+
+    @property
+    def stop(self):
+        return self.interval.sup
+
+    @property
+    def length(self):
+        return oo
+
+    def as_series(self, n=3):
+        """"returns the first n terms(non-zero) of the series
+        if n is none returns an iterator"""
+        if n is None:
+            return iter(self)
+
+        i, j = 0, 0
+        terms = []
+        while(i < n):
+                t = self.term(j)
+                if t != S.Zero:
+                    terms.append(t)
+                    i += 1
+                j += 1
+
+        return Add(*terms)
+
+    def shift(self, s):
+        """
+        Shift the function by a
+        term independent of x
+
+        f(x) -> f(x) + s
+
+        This is fast, if fourier series of f(x) is already
+        computed.
+
+        Examples
+        ========
+
+        >>> from sympy import FourierSeries, pi
+        >>> from sympy.abc import x
+        >>> s = FourierSeries(x**2, (x, -pi, pi))
+        >>> s.shift(1)
+        FourierSeries(x**2 + 1, (x, -pi, pi))
+        """
+        s, x = sympify(s), self.x
+
+        if x in s.free_symbols:
+            raise ValueError("'%s' should be independent of %s" %(s, x))
+
+        an = self.an
+        bn = self.bn
+        a0 = self.a0 + s
+        sfunc = self.args[0] + s
+
+        return FourierSeries(sfunc, self.args[1], a0=a0, an=an, bn=bn)
+
+    def shiftx(self, s):
```

Comment:
I'm not sure how necessary this is, can't a user just do a `.subs()` on the FourierSeries object?